### DISCUSSION TITLE
Generate a token per channel

### DISCUSSION THREAD

**Discussion by pablotdv** [datarhei/restreamer#490] [Q&A] [answered]
Is there a way to generate a token per channel?

Because the id generated to configure the stream platform is the same to use in the player.

My concern is about security as if someone knows a little more about the restreamer platform they can use the id to stream to a channel that belongs to another user.

So I would like to protect the stream URL with a token, but I want to generate a different token per channel.

**Comment by svenerbeck** [comment, marked as answer]
Hi @pablotdv 
Thank you for the ping. 🙈

A function comes in one of the following versions—a little patience. 

Cheerio
Sven
